_source/_posts/2022-02-25-thymeleaf-security.md

```diff
@@ -0,0 +1,670 @@
+---
+layout: blog_post
+title: "Securing your application when using Thymeleaf templates with Spring Webflux"
+author: jimena-garbarino
+by: contractor
+communities: [security,java]
+description: ""
+tags: [thymeleaf, security, method-security, oidc, webflux, spring, spring-boot, spring-security, csrf, authorization]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+
+Thymeleaf library has been around at least for 10 years and it is still actively maintained as of today. It is designed to allow stronger collaboration between design and developer teams for some use cases, as Thymeleaf templates look like HTML and can be displayed in the browser as static prototypes.
+In this post you will learn how to create a simple Spring Boot WebFlux application with Thymeleaf and Okta OIDC authentication, addressing the security concerns of preventing CSRF when submitting forms, and protecting functionality based on the user authorities and authentication status.
+
+
+**This tutorial was created with the following frameworks and tools**:
+
+- [HTTPie 3.0.2](https://httpie.io/)
+- [Java OpenJDK 11](https://jdk.java.net/java-se-ri/11)
+- [Okta CLI 0.9.0](https://cli.okta.com)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+# What is Thymeleaf?
+
+Thymeleaf is an open-source server-side Java template engine for standalone and web applications, that was created by Daniel Fernández. The templates look like HTML and can be integrated with Spring MVC and Spring Security among other popular development frameworks. Although documentation about integration with Spring WebFlux is hard to find, it is currently supported and the Thymeleaf starter dependency performs the [auto-configuration](https://docs.spring.io/spring-framework/docs/current/reference/html/web-reactive.html#webflux-view-thymeleaf) of the template engine, template resolver and reactive view resolver.
+
+Among its features, Thymeleaf allows to:
+
+- Work with fragments: only a part of a template can be rendered, which can be useful for updating part of a page from responses in AJAX calls. It also provides a componentization tool, as the fragments can be included in multiple templates.
```

Comment:
```suggestion
- Work with fragments: only a part of a template will be rendered. This can be useful for updating part of a page from responses in AJAX calls. It also provides a componentization tool, as the fragments can be included in multiple templates.
```